lib/internal/modules/esm/loader.js

```diff
@@ -332,9 +326,7 @@ class ESMLoader {
     if (this.#hooks) {
       loadResult = await this.#hooks.load(url, context);
     } else {
-      if (!this.#defaultLoad) {
-        this.#defaultLoad = require('internal/modules/esm/load').defaultLoad;
-      }
+      this.#defaultLoad ??= require('internal/modules/esm/load').defaultLoad;
```

Comment:
I agree these changed are good, but they're unrelated to this PR and conflict with changes in the off-thread PR.

Reply:
```suggestion
      if (!this.#defaultLoad) {
        this.#defaultLoad = require('internal/modules/esm/load').defaultLoad;
      }
```